feat(linux): DEV3702-ctu uboot - port newer NFS code in order to make…

doron2020 opened the pull request:
port nfs code from n366 uboot since it has a newer version of nfs (NfSv3). otherwise nfs will not work with this nfs server

notes:
I just copied nfs.c + nfs.h code from N366 uboot and fixed compilation errors
(e.g., different variable name - load_addr instead of image_load_address)

I tried to search the specific commit of Guillaume GARDET according to the message below but did not find it in git log of master (n366):

/* NOTE 4: NFSv3 support added by Guillaume GARDET, 2016-June-20.
 * NFSv2 is still used by default. But if server does not support NFSv2, then
 * NFSv3 is used, if available on NFS server. */

baruchsiach commented:
> port nfs code from n366 uboot since it has a newer version of nfs (NfSv3). otherwise nfs will not work with this nfs server
> 
> notes:
> I just copied nfs.c + nfs.h code from N366 uboot and fixed compilation errors
> (e.g., different variable name - load_addr instead of image_load_address)
> 
> I tried to search the specific commit of Guillaume GARDET according to the message below but did not find it in git log of master (n366):
> 
> /* NOTE 4: NFSv3 support added by Guillaume GARDET, 2016-June-20.
> 
>     * NFSv2 is still used by default. But if server does not support NFSv2, then
> 
>     * NFSv3 is used, if available on NFS server. */

You can use `git blame` to find that the commit that introduced this comment in `b0baca9820485`. But there is no need for that. Just mention in the commit log that the code is taken from U-Boot version 2020.04.

baruchsiach reviewed (changes requested):
Commit log needs update.

Commit:
feat: DEV-3702: ctu uboot - port newer NFS code in order to make nfs work

Port nfs code from n366 uboot since it has a newer version of nfs. otherwise nfs will not work with  the  ctu kernel version
This code is taken from U-Boot version 2020.04.
